streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -288,6 +288,22 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                 //  -> don't pass in `task.timeout.ms` to stay responsive if `KafkaStreams#close` gets called
                 final ConsumerRecords<byte[], byte[]> records = globalConsumer.poll(requestTimeoutPlusTaskTimeout);
```

Comment:
If we do not change this line, the restoring phase would be stuck for `requestTimeoutPlusTaskTimeout` (5m30s by default) for every partition not having any data to returned. Thus, potentially drastically increasing the required restore phase. Maybe we should invoke `poll()` more aggressively?

Reply:
Ah. Good point. We would need to refactor this anyway (cf the long comment above). Seems it's time to do it now :)